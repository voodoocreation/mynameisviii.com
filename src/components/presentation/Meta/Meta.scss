@import "../../../scss/mixins";

.Meta--label {
  @include hide-visually();
}
