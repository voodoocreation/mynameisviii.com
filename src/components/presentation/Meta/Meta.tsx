import cn from "classnames";
import * as React from "react";
import { FormattedMessage, InjectedIntlProps, injectIntl } from "react-intl";

import * as messages from "../../../locales/en-NZ";

import "./Meta.scss";

interface IProps extends InjectedIntlProps {
  className?: string;
  icon?: React.ReactNode;
  label?: React.ReactNode;
  labelIntlId?: keyof typeof messages;
  title?: string;
}

const Meta: React.FC<IProps> = ({
  children,
  className,
  icon,
  label,
  labelIntlId,
  title
}) => (
  <div className={cn("Meta", className)} title={title}>
    {icon}{" "}
    {labelIntlId || label ? (
      <span className="Meta--label">
        {labelIntlId ? <FormattedMessage id={labelIntlId} /> : label}:
      </span>
    ) : null}{" "}
    <span className="Meta--value">{children}</span>
  </div>
);

export default injectIntl(Meta);
