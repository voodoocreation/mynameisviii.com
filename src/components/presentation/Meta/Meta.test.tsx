import { render } from "enzyme";
import * as React from "react";

import Meta from "./Meta";

const setup = (fn: any, fromTestProps?: any) => {
  const props = {
    className: "TestMeta",
    ...fromTestProps
  };

  return {
    actual: fn(<Meta {...props}>Meta value</Meta>),
    props
  };
};

describe("[presentation] <Meta />", () => {
  it("renders correctly with `label` prop", () => {
    const { actual } = setup(render, { label: "Test label" });

    expect(actual).toMatchSnapshot();
  });

  it("renders correctly with `labelConstant` prop", () => {
    const { actual } = setup(render, { labelConstant: "GENRE" });

    expect(actual).toMatchSnapshot();
  });
});
