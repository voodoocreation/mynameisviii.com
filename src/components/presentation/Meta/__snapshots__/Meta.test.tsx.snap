// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <Meta /> renders correctly with \`label\` prop 1`] = `
<div
  class="Meta TestMeta"
>
   
  <span
    class="Meta-label"
  >
    Test label:
  </span>
   
  <span
    class="Meta-value"
  >
    Meta value
  </span>
</div>
`;

exports[`[presentation] <Meta /> renders correctly with \`labelConstant\` prop 1`] = `
<div
  class="Meta TestMeta"
>
   
  <span
    class="Meta-label"
  >
    <span>
      Genre
    </span>
    :
  </span>
   
  <span
    class="Meta-value"
  >
    Meta value
  </span>
</div>
`;
